Remove statement that data is moved when assigning (#982)
google/comprehensive-rust: src/ownership/move-semantics.md

```

* The assignment of `s1` to `s2` transfers ownership.
* When `s1` goes out of scope, nothing happens: it does not own anything.
* When `s2` goes out of scope, the string data is freed.
* There is always _exactly_ one variable binding which owns a value.

<details>

* Mention that this is the opposite of the defaults in C++, which copies by value unless you use `std::move` (and the move constructor is defined!).

> * It is only the ownership that moves. Whether the resulting machine code copies any data around is a matter of optimization, and such copies are aggressively optimized away.

Reviewer: The copy here is confusing again, right? 😬

Reviewer 2: Maybe this, although "manipulate" feels a bit awkward.
```suggestion
* It is only the ownership that moves. Whether any machine code is generated to manipulate the data itself is a matter of optimization, and such copies are aggressively optimized away.
```

Reviewer: I like it better

```

* The assignment of `s1` to `s2` transfers ownership.

> * When `s1` goes out of scope, nothing happens: it has no ownership.

Reviewer 2: ```suggestion
* When `s1` goes out of scope, nothing happens: it does not own anything.
```

* Mention that this is the opposite of the defaults in C++, which copies by value unless you use `std::move` (and the move constructor is defined!).

> * Data is not moved when you do `s2 = s1`, it is only the ownership that moves.

Reviewer 2: "Data .. moved" I think is referring to copy bytes here? I think this is a confusing way to use the word "move". And also, whether bytes get copied or not is actually about optimization, which is not what we're talking about here. Maybe change the emphasis:
```suggestion
* It is only the ownership that moves. Whether the resulting machine code copies any data around is a matter of optimization, and such copies are aggressively optimized away.
```
